10a-walkthrough-6.Rmd

```diff
@@ -0,0 +1,677 @@
+    filter(`SEA Disability Category` == "All Disabilities", 
+           `SEA Education Environment` %in% c("Total, Age 3-5", "Total, Age 6-21")) 
+```
+
+## Rename the variables
+
+In the next section we'll be preparing the dataset for visualization and modeling by "tidying" it. When we write code to transform datasets, we have to repeatedly type the column names so it's useful to change them to ones with reasonable lengths and without spaces. 
+
+```{r}
+child_counts <- child_counts %>% 
+    rename(year = Year,
+           state = "State Name", 
+           age = "SEA Education Environment",
+           disability = "SEA Disability Category", 
+           f_3_5 = "Female Age 3 to 5", 
+           m_3_5 = "Male Age 3 to 5", 
+           f_6_21 = "Female Age 6 to 21", 
+           m_6_21 = "Male Age 6 to 21")
+```
+
+## Clean state names
+
+You might have noticed that some state names in our dataset are in upper case letters and some are in lower case letters: 
+
+```{r}
+child_counts %>% 
+    count(state) %>% head()
+```
+
+If we leave it like this, R will treat the upper case states and lower case states of the same name as different states. We can use `mutate` and `tolower` to transform all the state names to lowercase letters. 
+
+```{r}
+child_counts <- child_counts %>% 
+    mutate(state = tolower(state)) 
+```
+
+## Tidy the dataset 
+
+Visualizing and modeling our data will be much easier if our dataset is in a "tidy" format. In his paper *Tidy Data*, Hadley Wickham defines tidy datasets where
+
+>1. Each variable forms a column.
+>2. Each observation forms a row.
+>3. Each type of observational unit forms a table.
+
+*A note on the gender variable in this dataset*
+
+*This dataset uses a binary approach to data collection about gender. Students are described as either male or female. The need for an inclusive approach to documenting gender identity is discussed in a document by Andrew Park of The Williams Institute at UCLA [authored a document](https://williamsinstitute.law.ucla.edu/wp-content/uploads/Reachable-Data-collection-methods-for-sexual-orientation-gender-identity-March-2016.pdf).*
+
+The gender variables in our dataset are spread across four columns, one representing a combination of gender and age range. We can use `gather` to bring the gender variable into one column. In this transformation, we create two new columns: a `gender` column and a `total` column. The `total` column will contain the number of students in each row's gender and age category.  
+
+```{r}
+child_counts <- child_counts %>% 
+    gather(gender, total, f_3_5:m_6_21)
+```
+
+To make the values of the `gender` column more intuitive, we'll use `case_when` to transform the values to either "f" or "m":
+
+```{r}
+child_counts <- child_counts %>% 
+    mutate(gender = case_when(
+        gender == "f_3_5" ~ "f", 
+        gender == "m_3_5" ~ "m", 
+        gender == "f_6_21" ~ "f", 
+        gender == "m_6_21" ~ "m", 
+        TRUE ~ as.character(gender)
```

Comment:
can we say just a little bit more about what this last line `TRUE ~ as.character(gender)` does? My understanding is it is always a true statement - although, it is evaluated following all of the other statements - and so it is basically saying, if gender isn't any of these values above, keep the original value for gender.